opentracing/global.go

```diff
@@ -0,0 +1,30 @@
+package opentracing
+
+import (
+	"golang.org/x/net/context"
+)
+
+var (
+	globalOpenTracer OpenTracer = noopOpenTracer{noopContextIDSource{}}
+)
+
+// Should be called as early as possible in main(), prior to calling the
+// `StartSpan*` (etc) global funcs below. Prior to calling `InitGlobal`, any
+// Spans started via the `StartSpan*` globals are noops.
+func InitGlobalTracer(rec Recorder, ctxIDSource ContextIDSource) {
```

Comment:
everything in this file can be avoided by people who think singletons are horrible :)